src/main/solution_space/path_finder.py

```diff
@@ -75,25 +72,26 @@ def __find_closest_vertex_with_path(self, user_code: Code, user: User, goal: Ver
         return self.__choose_best_vertex(user_code, user, candidates)
 
     @staticmethod
-    def __is_closer_goal(user_diffs_to_goal: int, user_diffs_to_graph: int) -> bool:
-        return user_diffs_to_graph / user_diffs_to_goal > TO_GRAPH_DISTANCE_THRESHOLD
+    def __is_far_from_graph(diffs_from_user_to_goal: int, diffs_from_user_to_graph_vertex: int) -> bool:
+        return diffs_from_user_to_graph_vertex / diffs_from_user_to_goal > DISTANCE_TO_GRAPH_THRESHOLD
 
     @staticmethod
-    def __is_almost_goal(empty_diffs_to_goal: int, user_diffs_to_goal: int) -> bool:
-        return abs(empty_diffs_to_goal - user_diffs_to_goal) <= empty_diffs_to_goal * DIFFS_PERCENT_GO_DIRECTLY
+    def __is_most_of_path_is_done(diffs_from_empty_to_goal: int, diffs_from_user_to_goal: int) -> bool:
+        return diffs_from_user_to_goal \
+               <= (diffs_from_empty_to_goal + diffs_from_user_to_goal) * DIFFS_PERCENT_TO_GO_DIRECTLY
```

Comment:
Sorry, but I think it's something wrong with this formula again.
shouldn't we sum diffs_from_empty_to_user and diffs_from_user_to_goal?